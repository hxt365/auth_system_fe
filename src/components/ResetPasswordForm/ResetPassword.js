import React from 'react';

function ResetPasswordForm() {
  return <p>ResetPasswordForm</p>;
}

export default ResetPasswordForm;
