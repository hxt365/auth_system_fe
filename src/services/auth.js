// @flow

import {
  LOGIN,
  LOGOUT,
  REFRESH,
  RESET_PASSWORD,
  CHANGE_PASSWORD,
  SIGNUP,
  CONFIRM_EMAIL,
} from 'constants/api';

import type {
  loginFormType,
  resetPasswordType,
  changePasswordFormType,
  signupFormType,
} from 'type';

import customAxios from './axios';

const login = async (data: loginFormType) => {
  const res = await customAxios.post(LOGIN, data).catch(err => err);
  return res;
};

const logout = async () => {
  const res = await customAxios.post(LOGOUT).catch(err => err);
  return res;
};

const resetPassword = async (data: resetPasswordType) => {
  const res = await customAxios.post(RESET_PASSWORD, data).catch(err => err);
  return res;
};

const refresh = async () => {
  const res = await customAxios.post(REFRESH).catch(err => err);
  return res;
};

const changePassword = async (data: changePasswordFormType) => {
  const res = await customAxios.post(CHANGE_PASSWORD, data).catch(err => err);
  return res;
};

const signup = async (data: signupFormType) => {
  const res = await customAxios.post(SIGNUP, data).catch(err => err);
  return res;
};

const confirmEmail = async (token: String) => {
  const link = `${CONFIRM_EMAIL + token}/`;
  const res = await customAxios.post(link).catch(err => err);
  return res;
};

export default {
  login,
  logout,
  refresh,
  resetPassword,
  changePassword,
  signup,
  confirmEmail,
};
