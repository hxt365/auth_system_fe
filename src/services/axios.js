// @flow

import axios from 'axios';

export const BASE_API_URL: String = `${process.env.REACT_APP_BASE_API_URL}`;

const customAxios = axios.create({
  baseURL: BASE_API_URL,
  withCredentials: true,
  xsrfCookieName: 'csrftoken',
  xsrfHeaderName: 'X-CSRFToken',
});

customAxios.interceptors.response.use(
  response => response,
  error => {
    // Add logic to handle error here
    console.log(error.response);
    return Promise.reject(error);
  },
);

export default customAxios;
