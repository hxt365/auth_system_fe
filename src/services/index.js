export { default as authServices } from './auth';
