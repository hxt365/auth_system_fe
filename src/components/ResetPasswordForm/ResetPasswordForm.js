// @flow

import React, { useEffect, useContext } from 'react';
import { Form, Icon, Button, Input } from 'antd';
import './ResetPasswordForm.scss';
import { authServices } from 'services';
import { hasErrors } from 'helpers/component';
import { AppContext } from 'components/AppLayout';
import { SUCCESS, ERROR } from 'constants/message';
import { HOME } from 'constants/route';
import { resetPasswordType } from 'type';

type PropsType = {
  form: any,
};

function ResetPasswordForm(props: PropsType) {
  const state = useContext(AppContext);

  useEffect(() => {
    props.form.validateFields();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const { form } = props;
  const { getFieldDecorator, getFieldsError, getFieldError, isFieldTouched } = form;

  const resetPassword = async (data: resetPasswordType) => {
    const res = await authServices.resetPassword(data);
    if (res.status === 200) {
      state.setMessage({
        status: SUCCESS,
        title: 'Please check your mail box!',
        button: 'Thank you, admin!',
        redirect: HOME,
      });
    } else {
      state.setMessage({
        status: ERROR,
        title: 'Please enter your own username and email!',
        button: 'Let me try again',
        redirect: '',
      });
    }
  };

  const handleSubmit = e => {
    e.preventDefault();
    props.form.validateFields((err, values) => {
      if (!err)
        resetPassword({
          username: values.username,
          email: values.email,
        });
    });
  };

  // Only show error after a field is touched.
  const usernameError = isFieldTouched('username') && getFieldError('username');
  const emailError = isFieldTouched('email') && getFieldError('email');
  return (
    <Form layout="inline" onSubmit={handleSubmit}>
      <Form.Item validateStatus={usernameError ? 'error' : ''} help={usernameError || ''}>
        {getFieldDecorator('username', {
          rules: [{ required: true, message: 'Please input your username!' }],
        })(
          <Input
            prefix={<Icon type="user" style={{ color: 'rgba(0,0,0,.25)' }} />}
            placeholder="Username"
          />,
        )}
      </Form.Item>
      <Form.Item validateStatus={emailError ? 'error' : ''} help={emailError || ''}>
        {getFieldDecorator('email', {
          rules: [
            { required: true, message: 'Please input your Email!' },
            { type: 'email', message: 'Your email is not valid' },
          ],
        })(
          <Input
            prefix={<Icon type="mail" style={{ color: 'rgba(0,0,0,.25)' }} />}
            type="email"
            placeholder="Email"
          />,
        )}
      </Form.Item>
      <Form.Item>
        <Button
          type="primary"
          onClick={handleSubmit}
          id="reset-form__button"
          disabled={hasErrors(getFieldsError())}
        >
          Get password
        </Button>
      </Form.Item>
    </Form>
  );
}

const wrapper = Form.create({ name: 'resetPasswordForm' })(ResetPasswordForm);

export default wrapper;
