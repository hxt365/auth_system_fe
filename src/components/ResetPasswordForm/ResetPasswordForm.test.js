import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import { authServices } from 'services';
import ResetPasswordForm from './ResetPasswordForm';

describe('test ResetPasswordForm', () => {
  it('test snapshot', () => {
    const wrapper = shallow(<ResetPasswordForm />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('submithandler called only once with username and email', () => {
    const resetPasswordSpy = jest.spyOn(authServices, 'resetPassword').mockImplementation(() => {});
    const wrapper = mount(
      <BrowserRouter>
        <ResetPasswordForm />
      </BrowserRouter>,
    );
    const usernameInput = wrapper.find('input#resetPasswordForm_username');
    expect(usernameInput).toHaveLength(1);
    const emailInput = wrapper.find('input#resetPasswordForm_email');
    expect(emailInput).toHaveLength(1);
    const username = 'username';
    const email = 'testing@example.com';
    usernameInput.simulate('change', { target: { value: username } });
    emailInput.simulate('change', { target: { value: email } });
    const btn = wrapper.find('button#reset-form__button');
    expect(btn).toHaveLength(1);
    btn.simulate('click');
    expect(resetPasswordSpy).toHaveBeenCalledTimes(1);
    expect(resetPasswordSpy).toHaveBeenCalledWith({
      username,
      email,
    });
    resetPasswordSpy.mockClear();
    wrapper.unmount();
  });

  it('submithandler not called because of empty username', () => {
    const resetPasswordSpy = jest.spyOn(authServices, 'resetPassword');
    const wrapper = mount(
      <BrowserRouter>
        <ResetPasswordForm />
      </BrowserRouter>,
    );
    const usernameInput = wrapper.find('input#resetPasswordForm_username');
    const emailInput = wrapper.find('input#resetPasswordForm_email');
    usernameInput.simulate('change', { target: { value: '' } });
    emailInput.simulate('change', { target: { value: 'testing@example.com' } });
    const btn = wrapper.find('button#reset-form__button');
    expect(btn).toHaveLength(1);
    btn.simulate('click');
    expect(resetPasswordSpy).toHaveBeenCalledTimes(0);
    resetPasswordSpy.mockClear();
    wrapper.unmount();
  });

  it('submithandler not called because of empty email', () => {
    const resetPasswordSpy = jest.spyOn(authServices, 'resetPassword');
    const wrapper = mount(
      <BrowserRouter>
        <ResetPasswordForm />
      </BrowserRouter>,
    );
    const usernameInput = wrapper.find('input#resetPasswordForm_username');
    const emailInput = wrapper.find('input#resetPasswordForm_email');
    usernameInput.simulate('change', { target: { value: 'username' } });
    emailInput.simulate('change', { target: { value: '' } });
    const btn = wrapper.find('button#reset-form__button');
    expect(btn).toHaveLength(1);
    btn.simulate('click');
    expect(resetPasswordSpy).toHaveBeenCalledTimes(0);
    resetPasswordSpy.mockClear();
    wrapper.unmount();
  });
});
