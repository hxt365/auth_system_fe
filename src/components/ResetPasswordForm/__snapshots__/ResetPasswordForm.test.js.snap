// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ResetPasswordForm test snapshot 1`] = `
<ResetPasswordForm
  form={
    Object {
      "getFieldDecorator": [Function],
      "getFieldError": [Function],
      "getFieldInstance": [Function],
      "getFieldProps": [Function],
      "getFieldValue": [Function],
      "getFieldsError": [Function],
      "getFieldsValue": [Function],
      "isFieldTouched": [Function],
      "isFieldValidating": [Function],
      "isFieldsTouched": [Function],
      "isFieldsValidating": [Function],
      "isSubmitting": [Function],
      "resetFields": [Function],
      "setFields": [Function],
      "setFieldsInitialValue": [Function],
      "setFieldsValue": [Function],
      "submit": [Function],
      "validateFields": [Function],
      "validateFieldsAndScroll": [Function],
    }
  }
/>
`;
